rotkehlchen/data/importer.py

```diff
@@ -934,3 +934,87 @@ def import_nexo_csv(self, filepath: Path) -> Tuple[bool, str]:
                 except KeyError as e:
                     return False, str(e)
         return True, ''
+
+    def _consume_shapeshift_trade(self, csv_row: Dict[str, Any]) -> None:
+        """
+        Consume the file containing only trades from ShapeShift.
+        """
+        timestamp = deserialize_timestamp_from_date(
+            date=csv_row['timestamp'],
+            formatstr='iso8601',
+            location='ShapeShift',
+        )
+        buy_asset = symbol_to_asset_or_token(csv_row['outputCurrency'])
+        buy_amount = deserialize_asset_amount(csv_row['outputAmount'])
+        sold_asset = symbol_to_asset_or_token(csv_row['inputCurrency'])
+        sold_amount = deserialize_asset_amount(csv_row['inputAmount'])
+        rate = deserialize_asset_amount(csv_row['rate'])
+        fee = deserialize_fee(csv_row['minerFee'])
+        in_addr = csv_row['inputAddress']
+        out_addr = csv_row['outputAddress']
+        # TODO: determine exact time ShapeShift started using SAI with trades in csv exports
+        sai_sunset_timestamp = deserialize_timestamp_from_date(
+            date='2019-11-18T00:00:00Z',
+            formatstr='iso8601',
+            location='ShapeShift',
+        )
+        notes = f'''
+Trade from ShapeShift with ShapeShift Deposit Address:
+ {csv_row['inputAddress']}, and
+ Transaction ID: {csv_row['inputTxid']}.
+  Refund Address: {csv_row['refundAddress']}, and
+ Transaction ID: {csv_row['refundTxid']}.
+  Destination Address: {csv_row['outputAddress']}, and
+ Transaction ID: {csv_row['outputTxid']}.
+'''
+        if sold_amount == ZERO:
+            log.debug(f'Ignoring ShapeShift trade with sold_amount equal to zero. {csv_row}')
+            return
+        if in_addr == '' or out_addr == '':
+            log.debug(f'Ignoring ShapeShift trade which was performed on DEX. {csv_row}')
+            return
+        # Assuming that as DAI contract tx date only SAI existed.
+        # Converting DAI to SAI in buy_asset and sell_asset.
+        if buy_asset == symbol_to_asset_or_token('DAI') and timestamp <= sai_sunset_timestamp:
```

Comment:
no need to use `symbol_to_asset_or_token()` here. This will end up searching the global database for something with a symbol "DAI". A user may also add another "DAI" symbol token and then we will have problems. (in that case the function raises.

What you want is to import `A_SAI` and `A_DAI` from constats/assets.py
https://github.com/rotki/rotki/blob/c4be84e692aa711a1cc7272d33d3209622ec40a0/rotkehlchen/constants/assets.py#L196